cellprofiler/modules/tram.py

```diff
@@ -0,0 +1,673 @@
+from cellprofiler.modules import trackobjects
+import numpy as np
+import itertools as it
+import cellprofiler.module as cpm
+import cellprofiler.setting as cps
+import cellprofiler.measurement as cpmeas
+from collections import Counter, defaultdict
+import re
+from cellprofiler.measurement import M_NUMBER_OBJECT_NUMBER
+import logging
+
+CAT_TRAM = "TrAM"
+MEAS_TRAM = "TrAM"
+MEAS_LABELS = "Labels"
+MEAS_PARENT = "Is_Parent"
+MEAS_SPLIT = "Split_Trajectory"
+FULL_TRAM_MEAS_NAME = "%s_%s" % (CAT_TRAM, MEAS_TRAM)
+FULL_LABELS_MEAS_NAME = "%s_%s" % (CAT_TRAM, MEAS_LABELS)
+FULL_PARENT_MEAS_NAME = "%s_%s" % (CAT_TRAM, MEAS_PARENT)
+FULL_SPLIT_MEAS_NAME = "%s_%s" % (CAT_TRAM, MEAS_SPLIT)
+IMAGE_NUM_KEY = "Image"
+MIN_TRAM_LENGTH = 6 # minimum number of timepoints to calculate TrAM
+MIN_NUM_KNOTS = 3
+
+LABELS_KEY = "labels"
+IMAGE_NUMS_KEY = "image_nums"
+OBJECT_NUMS_KEY = "object_nums"
+PARENT_OBJECT_NUMS_KEY = "parent_object_nums"
+TRAM_KEY = "TrAM"
+SPLIT_KEY = "split"
+PARENT_KEY = "parent"
+
+FLOAT_NAN = float('nan')
+
+__doc__ = """
+<b>TrAM</b> Provides a metric for tracking quality based on temporal
+smoothness of features measured across the trajectory (Tracking Aberration
+Measure).
+<hr>
+This module must be placed downstream of a module that identifies objects
+(e.g., <b>IdentifyPrimaryObjects</b>) and a <b>TrackObjects</b> that tracks
+them. There must be at least %d frames to perform a TrAM analysis. The TrAM
+statistic reflects how typical the maximum deviation from smooth time series
+a chosen set of measurements are. Typical fluctuations are determined from
+measurement differences in adjacent time points of objects whose trajectories
+are complete and without splitting events.
+
+<p><b>TODO</b>For an example pipeline using TrAM see the CellProfiler
+<a href="http://www.cellprofiler.org/examples.html#TrAM">Examples</a> webpage.</p>
```

Comment:
@mcquin @N3llz Some possible options on an example for `MeasureTrackQuality`:

1. One option would be to tack a `MeasureTrackQuality` node on the existing example for `TrackObjects`. Hopefully there is a range of track quality that TrAM would reveal. Are there plans to update this existing pipeline for the v3 release? It's loading with some errors since it's not up to date.

1. Another would be to upload the example from the book chapter we submitted. The image data are big, though, like >200MB. Multiple fluorophores. Could cut in half by examining just one of the two microscope fields.

1. Last option would be to delay uploading an example until after v3 is released.

Thoughts?